### PR TITLE
Create Journal Interaction

### PR DESCRIPTION
**Pull request by VinMoeMac**
## Description

Changed the code and put the form handler in modal-journal.js now takes in storage correctly
## How Has This Been Tested?

Live server
## Checklist
- [x] All automated tests have been passed locally (unit / e2e / linter / coverage)
- [x] New unit and E2E tests have been added to address the changes made
- [x] Manual testing has been performed according to the provided instructions
- [x] Code follows team's coding style, convention and standards
- [x] Documentation has been updated to reflect relevant changes (if applicable)

## Screenshots (if applicable):

### CODE REVIEW

File: src/styles/main.css [hunk 1]
```diff
@@ -1,3 +1,16 @@
 * {
     color: red;
-}
\ No newline at end of file
+}
```
Comment (jxnxthxnz) [inline src/styles/main.css:3]: move this into the modal-journal.js

File: src/scripts/database/stores/journal.js [hunk 2]
```diff
@@ -58,23 +58,26 @@ export function deleteJournal(path) {
...
+export function createJournal(title, path, tags, dueDate) {
     if (!!getJournal(path)) throw new Error(`Journal with path ${path} already exists!`);
 
     const journal = {
         title,
         path,
-        content,
+        content: "",
         tags,
+        dueDate,
```
Comment (nadecancode) [inline src/scripts/database/stores/journal.js:73]: A journal should not have due date, it's task's property isn't it?

File: src/scripts/modal-journal.js [hunk 3]
```diff
@@ -84,24 +86,52 @@ class ModalJournal extends HTMLElement {
...
     }
-}
+
+    //form handler to take data and put it in local storage
+    handleFormSubmit(event) {
+        event.preventDefault();
+        const formData = new FormData(event.target);
+        const journalData = {
+            title: formData.get('title'),
+            due: formData.get('due'),
```
Comment (nadecancode) [inline src/scripts/modal-journal.js:118]: I don't think we need due date when creating a journal

Comment (nadecancode) [reply src/scripts/modal-journal.js:118]: Instead, we ask user to link tasks when creating a journal

File: src/scripts/modal-journal.js [hunk 4]
```diff
@@ -84,24 +86,52 @@ class ModalJournal extends HTMLElement {
...
+            tags: formData.get('tags')
+        };
+        
+        createJournal(formData.get('title'),formData.get('path'),formData.get('tags'),formData.get('due'));
```
Comment (nadecancode) [inline src/scripts/modal-journal.js:122]: ^

**Review by nadecancode — CHANGES_REQUESTED** [review]

File: src/scripts/modal-journal.js [hunk 5]
```diff
@@ -84,24 +87,51 @@ class ModalJournal extends HTMLElement {
...
+        event.preventDefault();
+        const formData = new FormData(event.target);
+        
+        createJournal(formData.get('title'),formData.get('path'),formData.get('tags').split(", "));
+        let tasks=formData.get("tasks").split(", ");
+        tasks.forEach(element =>  {
+            linkTaskToJournal(element,getJournal(formData.get('path')));
+        });
+            // Handle the form data as needed
+        console.log('Journal created: ', getJournal(formData.get('path')));
```
Comment (nadecancode) [inline src/scripts/modal-journal.js:131]: Should be removed in production but I guess it's ok for now

**Review by nadecancode — APPROVED** [review]

**Pull request merged**